pkg/receive/handler_test.go

```diff
@@ -226,6 +226,15 @@ func TestDetermineWriteErrorCause(t *testing.T) {
 			threshold: 1,
 			exp:       errors.New("baz: 3 errors: 3 errors: qux; rpc error: code = AlreadyExists desc = conflict; rpc error: code = AlreadyExists desc = conflict; foo; bar"),
 		},
+		{
```

Comment:
I wonder why this issue was not caught by tests. Do we run unit tests for Ketama?

Reply:
We do have tests to verify quorum but they all run just with hashmod, I'm extending those as well to run with both algorithms.

Reply:
Are you referring to the e2e tests that were extended? Can we also do the same for unit tests?

Reply:
All relevant unit tests now have been expanded to run with both algorithms.

Reply:
Out of curiosity, did tests fail before your fix?

Reply:
Not really, which I believe was mainly because:
- With replication factor 3 and 3 nodes (which were all the test cases), even with ketama, it probably cannot be caught since the number of requests is small (but even on `main` expanding tests to simulate 6 node deployment, the failures start to show up
- We were running all tests with single timeseries, which is probably more of an edge case (now I run the tests with 50 series which should give us a good enough test data)
- The node addresses were randomized, I believe on `main` on ketama this was causing some incosistent results (I pushed fix for this now as well to generate the addresses in a predictable manner, no need for randomized addresses I believe)